docs/SPEC.md

```diff
@@ -2125,6 +2128,14 @@ from(bucket:"telegraf/autogen")
                 r.service == "app-server")
 ```
 
+Example:
+
+```
+from(bucket:"telegraf/autogen")
+    |> range(start:-12h)
+    |> filter(fn: (r) => r._measurement == "cpu", empty: "keep")
```

Comment:
What does `keep` mean? For example given the above query, should data be returned with `_measurement` equal to `mem`? It would be an empty table so should it be kept or dropped? 

Does keep mean: only keep tables with a group key that matches the predicate but could be empty? What if a predicate function applies to columns outside the group key, does that mean no empty tables would ever be kept?

Reply:
It was my opinion that the group key didn't matter. It had to do with the length of the table. If the table was empty, it was dropped. If it wasn't, then it was kept.

I'm not 100% sure I understand the question, but the above would return the full table if `_measurement` was equal to `cpu` and it would return an empty table for any series where `_measurement` was equal to anything else. So you would still get tables for all of the other series, but they would be empty.

Reply:
> So you would still get tables for all of the other series, but they would be empty.

This is not what storage does and seems like the wrong behavior. Does the user really want all series in their bucket when they set `empty: 'keep'`? That seems unlikely and not useful.

Reply:
Yes that is not what storage does. The default is proposed as `drop` which is the behavior that storage has if I understand correctly. @wolffcm did make a good point that this may not be true for value comparisons though which I hadn't thought of.

At the moment, the push down filter that influxdb uses has `empty: "drop"` as its behavior (potentially sans-value comparisons) and the flux filter has `empty: "keep"` as its behavior. This proposal is to make `drop` the default and to have the option to specify the behavior to allow for counting empty windows.